robot/Cumulus/resources/LeadsPageObject.py

```diff
@@ -0,0 +1,45 @@
+from cumulusci.robotframework.pageobjects import ListingPage
+from cumulusci.robotframework.pageobjects import DetailPage
+from cumulusci.robotframework.pageobjects import pageobject
+from cumulusci.robotframework.pageobjects import BasePage
+from BaseObjects import BaseNPSPPage
+from NPSP import npsp_lex_locators
+
+
+@pageobject("Custom", "Lead")
+class CovertLeadPage(BaseNPSPPage, BasePage):
+
+    def _is_current_page(self):
+        """
+        Waits for the Leads iframe to load
+        """
+        locator = '//iframe[contains(@name,"vfFrameId")]'
+        self.selenium.wait_until_element_is_visible(locator, timeout=60)
+        frame = self.selenium.driver.find_element_by_xpath(locator)
+        self.selenium.driver.switch_to.frame(frame);
+
+
+    def click_lead_convert_button(self):
+        self.npsp.page_scroll_to_locator('button', 'Convert')
+        self.selenium.click_button('Convert')
+
+
+@pageobject("Listing", "Lead")
+class LeadListingPage(BaseNPSPPage, ListingPage):
+    object_name = "Lead"
+
+    def click_delete_account_button(self):
+        """Clicks on Delete Account button inside the iframe"""
+        self.selenium.wait_until_location_contains("/list", message="Lead Listing page did not load in 30 seconds")
+
+
+@pageobject("Details", "Lead")
+class LeadDetailPage(BaseNPSPPage, DetailPage):
+    object_name = "Lead"
+
+    def _is_current_page(self):
+        """ Verify we are on the Lead detail page
+            by verifying that the url contains '/view'
+        """
+        self.selenium.wait_until_location_contains("/view", timeout=60, message="Detail page did not load in 1 min")
```

Comment:
Do we need to hard code timeouts? Did you try without specifying timeouts?

Reply:
I tried locally and it works fine but on the Metaci, I felt like a timeout is required.

Reply:
We usually avoid adding hardcoded timeouts unless it is really required. So did you test this metaci. What is the behavior in Metaci? Did it not wait until the url was there?

Reply:
Yes on MetaCi, this was needed, we added this to all our page objects. I was just following the fix placed earlier

Reply:
yes on metaci occassionally pages took longer to load and default 30 secs was not suffiecient so we increased it to 60 to be on safe side and if the page loads quickly it wont wait for entire 60 seconds anyway